staticd/static_vty.c

```diff
@@ -425,24 +425,18 @@ static int static_route_leak(
 	if (table_str)
 		table_id = atol(table_str);
 
-	/* Null0 static route.  */
-	if (ifname != NULL) {
-		if (strncasecmp(ifname, "Null0", strlen(ifname)) == 0
-		    || strncasecmp(ifname, "reject", strlen(ifname)) == 0
-		    || strncasecmp(ifname, "blackhole", strlen(ifname)) == 0) {
+	/* Route flags */
+	if (flag_str) {
+		if (strncmp(flag_str, "NULL0", strlen(flag_str)) == 0 ||
+		    strncmp(flag_str, "null0", strlen(flag_str)) == 0) {
```

Comment:
`strcasecmp`?

Reply:
I wanted to accept Null0 without a warning.  Hence the reason it is done this way

Reply:
This is still wrong, we have to use the compare functions without the `n` version.

```c
#include <stdio.h>
#include <string.h>
#include <strings.h>

int
main(int argc, char *argv[])
{
        const char *user_input = "NuLl01";
        char expected[64];
        size_t expected_len;
        int scc, sc;

        strcpy(expected, "null0");
        expected_len = strlen(expected);
        sc = strcasecmp(expected, user_input);
        scc = strncasecmp(expected, user_input, expected_len);
        printf("strcasecmp(%s, %s, %ld) == %d\n",
            expected, user_input, expected_len, sc);
        printf("strncasecmp(%s, %s, %ld) == %d\n",
            expected, user_input, expected_len, scc);

        return 0;
}
```

Prints:

```
strcasecmp(null0, NuLl01, 5) == -49
strncasecmp(null0, NuLl01, 5) == 0
```

If you use the `n` version and use the length of the source string you won't get what you expect. The `n` version serves the purpose of comparing at *most* `N` items.

This is related with issue #4716 .